reactor-netty-core/src/main/java/reactor/netty/channel/FluxReceive.java

```diff
@@ -153,11 +150,12 @@ public void subscribe(CoreSubscriber<? super Object> s) {
 	}
 
 	final void startReceiver(CoreSubscriber<? super Object> s) {
-		if (once == 0 && ONCE.compareAndSet(this, 0, 1)) {
+		if (!subscribedOnce) {
+			subscribedOnce = true;
 			if (log.isDebugEnabled()) {
 				log.debug(format(channel, "{}: subscribing inbound receiver"), this);
 			}
-			if (inboundDone && getPending() == 0) {
+			if ((inboundDone && getPending() == 0) || isCancelled()) {
```

Comment:
since cancelation is possible when there is a subscriber, in that case isCancelled would mean dispose happened earlier